dds_cli/project_info.py

```diff
@@ -0,0 +1,118 @@
+import datetime
+import rich
+import sys
+
+# Own modules
+from dds_cli import base
+from dds_cli import exceptions
+from dds_cli import DDSEndpoint
+import dds_cli.utils
+
+###############################################################################
+# START LOGGING CONFIG ################################# START LOGGING CONFIG #
+###############################################################################
+
+LOG = logging.getLogger(__name__)
+
+
+###############################################################################
+# CLASSES ########################################################### CLASSES #
+###############################################################################
+
+
+class ProjectInfoManager(base.DDSBaseClass):
+    """Project info manager class."""
+
+    def __init__(
+        self,
+        project: str,
+        no_prompt: bool = False,
+        token_path: str = None,
+    ):
+        """Handle actions regarding project info in the cli."""
+        # Initiate DDSBaseClass to authenticate user
+        super().__init__(
+            no_prompt=no_prompt,
+            method_check=False,
+            token_path=token_path,
+        )
+        self.project = project
+
+    # Public methods ###################### Public methods #
+    def show_project_info(self):
+        """Get a project info."""
+        # Get info about a project from API
+        response, _ = dds_cli.utils.perform_request(
+            DDSEndpoint.PROJ_INFO,
+            method="get",
+            headers=self.token,
+            params={"project": self.project},
+            error_message="Failed to get project information",
+        )
+
+        project_info = response.get("project_info")
+
+        # Print project info table
+        table = dds_cli.utils.create_table(
+            title="Project information.",
+            columns=["Project ID", "Created by", "Status", "Last updated", "Size"],
+            rows=[
+                project_info,
+            ],
+            caption=f"Information about project {project_info['Project ID']}",
+        )
+        dds_cli.utils.console.print(table)
+
+        # Print Title and Description below the table
+        dds_cli.utils.console.print(f"[b]Project title:[/b]       {project_info['Title']}")
+        dds_cli.utils.console.print(f"[b]Project description:[/b] {project_info['Description']}")
+
+    def update_info(self, title=None, description=None, pi=None):
+        """Update project info"""
+
+        # Collect the items for change and print them before asking for confirmation
+        info_items = {}
+        if title:
+            info_items["title"] = title
+            dds_cli.utils.console.print(f"[b]New project title:[/b]         {info_items['title']}")
+        if description:
+            info_items["description"] = description
+            dds_cli.utils.console.print(
+                f"[b]New project description:[/b]   {info_items['description']}"
+            )
+        if pi:
```

Comment:
```suggestion
        if pi:
            # get project info pi and confirm that they want to change to the specified

```

Reply:
this is not going to work unless we change the get project info endpoint so that it returns PI (not returned at the moment)

Reply:
Aah ok! Yeah then we should do that.

Reply:
Maybe that'll make this PR too big since we need to fix the tests etc. We can discuss it tomorrow and maybe I can do a PR for only that first.

Reply:
it seems just adding PI in the response does not break things